### PR TITLE
Use outcode country

### PR DESCRIPTION
**Pull request by neilbmclaughlin**
Also, moved code around to make clear distinction between validation and rendering.

The unit tests for the middleware are just a start at this point just addressing testing error condition handling for dependent services.

### CODE REVIEW

**Comment by c2s-dev** [comment]
:rocket: deployment of gp-finder succeeded (http://gp-finder-pr-164.dev.beta.nhschoices.net)

File: app/middleware/postcodeValidator.js [hunk 1]
```diff
@@ -1,48 +1,42 @@
...
-    log.info('validate-location-end');
+    log.info('validate-postcode-end');
   }
 }
 
-function validateLocation(req, res, next) {
+module.exports = (req, res, next) => {
   if (res.locals.postcode) {
-    if (isNotEnglishPostcode(res.locals.postcode)) {
-      postcodeValidator.renderPostcodeNotEnglish(res.locals.postcode, req, res);
```
Comment (beseven) [inline app/middleware/postcodeValidator.js]: One issue I have with this is that for a full out of England postcode (that starts with the regex) we are doing to calls to the API now, while before we were filtering it out at start. 

If the overhead is acceptable for less code, than it's fine by me.

File: app/lib/postcodeValidator.js [hunk 2]
```diff
@@ -12,39 +8,7 @@ function isPostcode(postcode) {
   return postcode.match(postcodeRegex);
 }
 
-function setInvalidPostcodeLabel(res, postcode) {
```
Comment (beseven) [inline app/lib/postcodeValidator.js]: I agree taking rendering out of lib.

File: app/middleware/postcodeLookup.js [hunk 3]
```diff
@@ -1,26 +1,35 @@
...
         next();
       } else if (!err) {
         log.info('validate-outcode-invalid');
-        postcodeValidator.renderInvalidPostcodePage(postcode, req, res);
+        renderer.invalidPostcodePage(postcode, req, res);
         log.info('validate-outcode-end');
       } else {
         log.info('lookup-outcode-error');
-        postcodeValidator.handlePostcodeError(err, outcodeDetails, res, next);
+        renderer.handlePostcodeError(err, outcodeDetails, res, next);
```
Comment (beseven) [inline app/middleware/postcodeLookup.js]: I thought this method got renamed to `postcodeError `

File: app/middleware/postcodeLookup.js [hunk 4]
```diff
@@ -31,12 +40,12 @@ function lookupPostcode(req, res, next) {
...
-        log.info('revalidate-postcode-notEnglish');
-        postcodeValidator.renderPostcodeNotEnglish(postcodeDetails, req, res);
-        log.info('revalidate-postcode-notEnglish');
+        log.info('validate-postcode-notEnglish');
+        renderer.postcodeNotEnglish(postcode, req, res);
+        log.info('validate-postcode-notEnglish');
       } else {
         log.info('lookup-postcode-error');
-        postcodeValidator.handlePostcodeError(err, postcodeDetails, res, next);
+        renderer.handlePostcodeError(err, postcodeDetails, res, next);
```
Comment (beseven) [inline app/middleware/postcodeLookup.js]: I thought this method got renamed to `postcodeError `

File: app/middleware/postcodeLookup.js (hunk 3 above)
Comment (neilbmclaughlin) [inline app/middleware/postcodeLookup.js]: This suggests that these branches of code are not covered by current testing. I'll take a look and add some tests then fix.

**Comment by c2s-dev** [comment]
:rocket: deployment of gp-finder succeeded (http://gp-finder-pr-164.dev.beta.nhschoices.net)

**Comment by c2s-dev** [comment]
:rocket: deployment of gp-finder succeeded (http://gp-finder-pr-164.dev.beta.nhschoices.net)

File: app/middleware/postcodeValidator.js (hunk 1 above)
Comment (neilbmclaughlin) [inline app/middleware/postcodeValidator.js]: I think it's premature optimisation with added code complexity so I'd say it's worth it.

**Comment by neilbmclaughlin** [comment]
@beseven - tests added and fixes made. Should be good for merge.

**Comment by beseven** [comment]
LGTM!

**Pull request merged**